openedx_tagging/core/tagging/rules.py

```diff
@@ -65,6 +65,44 @@ def can_change_tag(user: UserType, tag: Tag | None = None) -> bool:
     )
 
 
+@rules.predicate
+def can_view_object_tag_objectid(_user: UserType, _object_id: str) -> bool:
+    """
+    Everybody can view object tags from any objects.
+
+    This rule could be defined in other apps for proper permission checking.
+    """
+    return True
+
+
+@rules.predicate
+def can_view_object_tag(
+    user: UserType, perm_obj: ObjectTagPermissionItem | None = None
+) -> bool:
+    """
+    Checks if the user has permissions to view tags on the given taxonomy and object_id.
+    """
+
+    # The following code allows METHOD permission (GET) in the viewset for everyone
+    if perm_obj is None:
+        return True
+
+    # Checks the permission for the taxonomy
+    taxonomy_perm = user.has_perm(
+        "oel_tagging.view_objecttag_taxonomy", perm_obj.taxonomy
```

Comment:
I'm unclear why we have a separate `view_objecttag_taxonomy` permission here. Is there any reason that would ever be different from just the `oel_tagging.view_taxonomy` permission? I think the code should just check `oel_tagging.view_taxonomy` here, and we can delete the redundant `view_objecttag_taxonomy` permission, unless I'm missing something.

After all, there's not even any data about the objecttag being passed into the permissions check, so it really is just checking "can user view the taxonomy" not "can user view the taxonomy associate with this object tag".

Reply:
There is only a minor difference in the edx-platform: admins can see disabled taxonomies, but can't see ObjectTags using disabled taxonomies.
I removed (https://github.com/openedx/edx-platform/pull/33413/commits/381f4dfc987cdadd36a04012267f5c60684189a3) from `edx-platform` I implemented here (6b9198029ce64be1e1f4af2b106226867b35209d)

We also mirrored the `change_objecttag` permission, having separated objectid and taxonomy permissions.

Reply:
Done: bc8efc0550077972bd311aa9f7d3caec53e724a8

I also fixed some tests. Waiting CD/CI results!